plugin/storage/cassandra/spanstore/operation_names.go

```diff
@@ -51,13 +80,23 @@ func NewOperationNamesStorage(
 	metricsFactory metrics.Factory,
 	logger *zap.Logger,
 ) *OperationNamesStorage {
+
+	schemaVersion := LatestVersion
+
+	if !tableExist(session, schemas[schemaVersion].TableName) {
+		schemaVersion = schemaVersion - 1
+	}
+
 	return &OperationNamesStorage{
-		session:       session,
-		InsertStmt:    insertOperationName,
-		QueryStmt:     queryOperationNames,
-		metrics:       casMetrics.NewTable(metricsFactory, "operation_names"),
-		writeCacheTTL: writeCacheTTL,
-		logger:        logger,
+		session:         session,
+		TableName:       schemas[schemaVersion].TableName,
+		SchemaVersion:   schemaVersion,
+		InsertStmt:      fmt.Sprintf(schemas[schemaVersion].InsertStmt, schemas[schemaVersion].TableName),
+		QueryByKindStmt: fmt.Sprintf(schemas[schemaVersion].QueryByKindStmt, schemas[schemaVersion].TableName),
+		QueryStmt:       fmt.Sprintf(schemas[schemaVersion].QueryStmt, schemas[schemaVersion].TableName),
```

Comment:
in line with the other comments, this could be a single field, created like this:
```
table: schemas[schemaVersion].materialize()
```

All sprintf can be done in materialize(), as well as creating the respective insert/query functions.